pkgs/tools/typesetting/tectonic/default.nix

```
@@ -16,20 +16,37 @@
 , openssl
 , pkg-config
 , icu
+, fetchpatch2
+, applyPatches
 }:
 
 rustPlatform.buildRustPackage rec {
   pname = "tectonic";
   version = "0.15.0";
 
-  src = fetchFromGitHub {
-    owner = "tectonic-typesetting";
-    repo = "tectonic";
-    rev = "tectonic@${version}";
-    sha256 = "sha256-xZHYiaQ8ASUwu0ieHIXcjRaH06SQoB6OR1y7Ok+FjAs=";
+  /**
+    `src` is patched up front before Cargo vendoring. We cannot reply on `patchPhase`
+    since Cargo vendoring happens before `patchPhase`. This is documented in:
+
+    https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md
+  */
+  src = applyPatches {
+    src = fetchFromGitHub {
+      owner = "tectonic-typesetting";
+      repo = "tectonic";
+      rev = "tectonic@${version}";
+      sha256 = "sha256-xZHYiaQ8ASUwu0ieHIXcjRaH06SQoB6OR1y7Ok+FjAs=";
+    };
+    patches = [
+      /** patch Cargo.lock to fix build with rust 1.80 */
```

```suggestion
      # fix build with rust 1.80
```
using multiline comments here is unecessary.

and comments should not describe the obvious but what it does, similar to git commit messages.

---

> describe the obvious

From the url it is not obvious that we are patching the Cargo.lock file, but whatever, I don't mind this, committing...